aws/cloudtrail.go

```diff
@@ -0,0 +1,64 @@
+package aws
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudtrail"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/cloud-nuke/logging"
+	"github.com/gruntwork-io/go-commons/errors"
+)
+
+func getAllCloudtrailTrails(session *session.Session, excludeAfter time.Time, configObj config.Config) ([]*string, error) {
+	svc := cloudtrail.New(session)
+
+	param := &cloudtrail.ListTrailsInput{}
+
+	trailIds := []*string{}
+
+	paginator := func(output *cloudtrail.ListTrailsOutput, lastPage bool) bool {
+		for _, trailInfo := range output.Trails {
+			trailIds = append(trailIds, trailInfo.TrailARN)
+		}
+		return !lastPage
+	}
+
+	err := svc.ListTrailsPages(param, paginator)
+	if err != nil {
+		return trailIds, errors.WithStackTrace(err)
+	}
+
+	return trailIds, nil
+}
+
+func nukeAllCloudTrailTrails(session *session.Session, arns []*string) error {
```

Comment:
Make sense to return `error` if each time function returns nil?

Reply:
Now I'm wondering what's best here. If we merge the PR ( #357 ) about improving the table report, this might become a moot issue.